app/services/github_service.rb

```diff
@@ -129,12 +131,27 @@ def full_repo_name
       def create_new_tree_for_post(client, post_markdown, post_title, sha_base_tree)
         # This blob represents the content we're going to create which in this case is markdown
         blob_sha = client.create_blob(full_repo_name, post_markdown)
-        client.create_tree(full_repo_name, 
-                          [ { path: "_posts/#{DateTime.now.strftime('%Y-%m-%d')}-#{post_title.gsub(/\s+/, '')}.md",
-                              mode: '100644',
-                              type: 'blob',
-                              sha: blob_sha } ],
-                             base_tree: sha_base_tree)[:sha]
+        blob_information = [ { path: "_posts/#{DateTime.now.strftime('%Y-%m-%d')}-#{post_title.gsub(/\s+/, '')}.md",
+                               mode: '100644',
+                               type: 'blob',
+                               sha: blob_sha } ]
+        create_image_blobs(client, blob_information, post_markdown)
+        client.create_tree(full_repo_name, blob_information, base_tree: sha_base_tree)[:sha]
+      end
+
+      def create_image_blobs(client, blob_information, post_markdown)
+        PostImageManager.instance.uploaders.each do |uploader|
+          # This check prevents against images that have been removed from the markdown
+          if KramdownService.does_markdown_include_image(uploader.filename, post_markdown)
+            # This line uses .file.file since the first .file returns a carrierware object
+            base_64_encoded_image = Base64.encode64(File.open(uploader.post_image.file.file, 'rb').read)
+            image_blob_sha = client.create_blob(full_repo_name, base_64_encoded_image, 'base64')
+            blob_information << { path: "assets/img/#{uploader.filename}",
+                                  mode: '100644',
```

Comment:
What is the meaning of mode "100644"?

Reply:
Good question, I didn't know the meaning of that mode either. I googled it and the 100644 stands for blob file storage for a Git tree. I added a comment explaining that and the GitHub tree API reference source here: https://developer.github.com/v3/git/trees/

Reply:
Ohh okay. Makes sense--thanks for the update and leaving a comment.